core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -62,6 +71,52 @@ oppia.directive('supplementalCard', [
             $scope.helpCardHasContinueButton = false;
           };
 
+          $scope.isCurrentHintUsable = false;
+
+          $scope.allHintsExhausted = false;
+
+          $scope.promiseForHint = null;
+
+          $scope.hintButtonClicks = 0;
+
+          $scope.tooltipText = '';
+
+          $scope.showHint = function() {
+            var hints = $scope.currentInteractionHints;
+            if ($scope.hintButtonClicks < hints.length) {
+              var currentHint = hints[$scope.hintButtonClicks].hintText;
+              playerTranscriptService.addNewAnswer('I would like a hint.');
+              playerTranscriptService.addNewFeedback(currentHint);
+              $scope.helpCardHtml = currentHint;
+              $scope.isCurrentHintUsable = false;
+              $scope.promiseForHint = $timeout(function() {
+                $scope.isCurrentHintUsable = true;
+              }, WAIT_FOR_HINT_MSEC);
+
+              if ($scope.hintButtonClicks === hints.length - 1) {
+                $scope.isCurrentHintUsable = false;
+                $scope.allHintsExhausted = true;
+                $scope.hintButtonClicks = 0;
+              } else {
+                $scope.hintButtonClicks += 1;
+              }
+            }
+          };
+
+          $scope.isHintAvailable = function() {
+            var hintIsAvailable = (
+              $scope.isCurrentHintUsable && !$scope.allHintsExhausted);
+            if ($scope.allHintsExhausted) {
+              $scope.tooltipText = 'Sorry, I am out of hints!';
+            } else if (!hintIsAvailable) {
+              $scope.tooltipText = (
+                'Try thinking a bit more before asking for another hint!');
+            } else {
+              $scope.tooltipText = 'Click here for a Hint!';
+            }
+            return hintIsAvailable;
+          };
```

Comment:
Could you refactor it such that this is just a getter method as per the name? Especially since this is in a binding that gets called repeatedly, let's try moving all the setters elsewhere and have them change/set only when they need to.